third_party/openthread/BUILD.gn

```diff
@@ -0,0 +1,25 @@
+import("//build_overrides/chip.gni")
+import("//build_overrides/openthread.gni")
+
+import("${chip_root}/src/platform/device.gni")
+
+declare_args() {
+  # Use a different target, such as a prebuilt Thread library.
+  chip_openthread_target = ""
+}
+
+group("openthread_device_deps") {
+  if (chip_openthread_target != "") {
+    public_deps = [ chip_openthread_target ]
+  } else {
+    public_deps = [
```

Comment:
Should -platform not be split out? It's a different library.

Reply:
I do believe you end up needing ftd or mtd either way, I can split it but then you need to add the deps elsewhere.

correct me if I am wrong. But it doesn't hurt to have them grouped

Reply:
I will split them and test it out in my next PR this week including the other work mentionned for the cli deps